### PR TITLE
chore(specReporter): enable testing via mocha

### PR DESCRIPTION
**Pull request by tmcgee123**
Also, add ability to generate coverage report for tests.
Added basic unit test for starting point.

This is in reference to #36. @hdavidzhu,  please feel free to add any comments/concerns. This is my first Mocha setup I've ever done.

### CODE REVIEW

**Comment by hdavidzhu** [comment]
that was fast, thanks!

File: package.json [hunk 1]
```diff
@@ -16,10 +17,16 @@
...
   "peerDependencies": {
     "karma": ">=0.9"
   },
-  "license": "MIT"
+  "license": "MIT",
+  "devDependencies": {
+    "chai": "^3.4.0",
+    "istanbul": "^0.4.0",
+    "mocha": "^2.3.3",
+    "mocha-runner": "^1.1.1"
```
Comment (hdavidzhu) [inline package.json]: do we still need mocha and chai with mocha-runner?

**Comment by hdavidzhu** [comment]
just one comment, good to merge afterwards. I just ran this locally and we are happy :smile:

**Comment by tmcgee123** [comment]
I was able to remove the mocha dependency, but since I `require('chai')` in the test I need it to be in package.json.

**Comment by hdavidzhu** [comment]
:+1: merge when ready

**Pull request merged**